src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion.py

```diff
@@ -77,60 +83,44 @@ def prepare_inputs(self, prompt: Union[str, List[str]]):
         )
         return text_input.input_ids
 
-    def __call__(
+    def _get_safety_scores(self, features, params):
+        special_cos_dist, cos_dist = self.safety_checker(features, params)
+        return (special_cos_dist, cos_dist)
+
+    def _run_safety_checker(self, images, safety_model_params, jit=False):
+        # safety_model_params should already be replicated when jit is True
+        pil_images = [Image.fromarray(image) for image in images]
+        features = self.feature_extractor(pil_images, return_tensors="np").pixel_values
+
+        if jit:
+            features = shard(features)
+            special_cos_dist, cos_dist = _p_get_safety_scores(self, features, safety_model_params)
+            special_cos_dist = unshard(special_cos_dist)
+            cos_dist = unshard(cos_dist)
+            safety_model_params = unreplicate(safety_model_params)
```

Comment:
why do we need to do `unreplicate` here ?

Reply:
Because if we are using `jit`, `safety_model_params` is extracted from the `params` dict which is already replicated. We use the replicated version in `_p_get_safety_scores` a couple of lines above, but then we need the unreplicated one to compute the scores in `self.safety_checker.filtered_with_scores`